src/structures/Chat.js

```diff
@@ -165,29 +165,43 @@ class Chat extends Base {
      * Loads chat messages, sorted from earliest to latest.
      * @param {Object} searchOptions Options for searching messages. Right now only limit is supported.
      * @param {Number} [searchOptions.limit=50] The amount of messages to return. Note that the actual number of returned messages may be smaller if there aren't enough messages in the conversation. Set this to Infinity to load all messages.
+	 * @param {Number} [searchOptions.timestamp] Unix timestamp to return only messages after a given date/time. If you do not provide a valid timestamp, it will be ignored; and will use only searchOptions.limit value
      * @returns {Promise<Array<Message>>}
      */
     async fetchMessages(searchOptions) {
-        if (!searchOptions || !searchOptions.limit) {
-            searchOptions = { limit: 50 };
+        let {limit, timestamp} = searchOptions || {};
+
+        if(timestamp < 1) timestamp = null;
+        if(limit === undefined || limit === null) limit = (timestamp ? limit : 50);
+        if(timestamp){
+            const date = new Date(timestamp * 1000);
+            if(date instanceof Date === false || isNaN(date.valueOf()))
+                timestamp = null;
         }
-        let messages = await this.client.pupPage.evaluate(async (chatId, limit) => {
-            const msgFilter = m => !m.isNotification; // dont include notification messages
+
+        let messages = await this.client.pupPage.evaluate(async (chatId, limit, timestamp) => {
+            // dont include notification messages... and if has a valid timestamp, add it to filter too
+            const msgFilter = m => !m.isNotification && (timestamp ? m.t > timestamp : true);
 
             const chat = window.Store.Chat.get(chatId);
             let msgs = chat.msgs.models.filter(msgFilter);
 
-            while (msgs.length < limit) {
+            let filteredMessages = [];
+            do{
                 const loadedMessages = await chat.loadEarlierMsgs();
                 if (!loadedMessages) break;
-                msgs = [...loadedMessages.filter(msgFilter), ...msgs];
-            }
+
+                filteredMessages = loadedMessages.filter(msgFilter);
+                msgs = [...msgs, ...filteredMessages];
```

Comment:
Why was this order reversed?

Reply:
Hello @pedroslopez, the idea is that the most recent messages are being taken up to a limit, it made sense for me to order them in a descending order.